pkg/hoist/hoist_launchable.go

```diff
@@ -374,10 +379,22 @@ func (hl *Launchable) Executables(
 				ExtraEnv:         map[string]string{launch.EntryPointEnvVar: relativePath},
 				NoLimits:         hl.ExecNoLimit,
 				CgroupConfigName: hl.CgroupConfigName,
-				PodID:            &hl.PodID,
 				CgroupName:       hl.CgroupName,
 				RequireFile:      hl.RequireFile,
 			}
+			if *IncludePodIDArg {
+				p2ExecArgs = p2exec.P2ExecArgs{
+					Command:          []string{filepath.Join(hl.InstallDir(), relativePath)},
+					User:             hl.RunAs,
+					EnvDirs:          []string{hl.PodEnvDir, hl.EnvDir()},
+					ExtraEnv:         map[string]string{launch.EntryPointEnvVar: relativePath},
+					NoLimits:         hl.ExecNoLimit,
+					CgroupConfigName: hl.CgroupConfigName,
+					PodID:            &hl.PodID,
+					CgroupName:       hl.CgroupName,
+					RequireFile:      hl.RequireFile,
+				}
+			}
```

Comment:
Could you modify p2ExecArgs PodID member with something like ```p2ExecArgs.PodID = &hl.PodID``` over declaring everything again? Otherwise change looks good.